pageserver/ctl/src/main.rs

```diff
@@ -19,42 +18,84 @@ use pageserver::{
     virtual_file,
 };
 use postgres_ffi::ControlFileData;
+use std::{
+    path::{Path, PathBuf},
+    str::FromStr,
+};
 use utils::{lsn::Lsn, project_git_version};
 
 project_git_version!(GIT_VERSION);
 
-const METADATA_SUBCOMMAND: &str = "metadata";
+#[derive(Parser)]
+#[command(
+    version = GIT_VERSION,
+    about = "Neon Pageserver binutils",
+    long_about = "Reads pageserver (and related) binary files management utility"
+)]
+#[command(propagate_version = true)]
+struct CliOpts {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    Metadata(MetadataCmd),
+    PrintLayerFile(PrintLayerFileCmd),
+    DrawTimeline {},
+    AnalyzeLayerMap(AnalyzeLayerMapCmd),
+    #[command(subcommand)]
+    Layer(LayerCmd),
+}
+
+/// Read and update pageserver metadata file
+#[derive(Parser)]
+struct MetadataCmd {
+    /// Input metadata file path
+    metadata_path: PathBuf,
+    /// Replace disk consistent Lsn
+    disk_consistent_lsn: Option<String>,
```

Comment:
Is there a benefit to parsing these late instead of at args parsing stage with `clap::Parser`?

As in, why not `Option<Lsn>`?

Reply:
This is copy-pasted from original code 🤪 I'm planning to change it to `Option<Lsn>`.